api/client/src/main/java/org/projectnessie/client/http/HttpClientResponseException.java

```diff
@@ -15,16 +15,56 @@
  */
 package org.projectnessie.client.http;
 
+import java.net.URI;
+import java.util.Optional;
+
+/**
+ * A generic exception thrown when the server replies with an unexpected status code.
+ *
+ * <p>This exception is generally thrown when the status code is not in the 2xx or 3xx range, and if
+ * a response filter did not throw a more specific exception.
+ */
 public class HttpClientResponseException extends HttpClientException {
 
+  private final URI uri;
   private final Status status;
+  private final String body;
 
-  public HttpClientResponseException(String message, Status status) {
+  public HttpClientResponseException(URI uri, Status status, String body) {
+    this(createMessage(uri, status, body), uri, status, body);
+  }
+
+  public HttpClientResponseException(String message, URI uri, Status status, String body) {
     super(message);
+    this.uri = uri;
     this.status = status;
+    this.body = body;
+  }
+
+  public URI getUri() {
```

Comment:
Don't see a production code usage of this getter.

Reply:
There isn't; it's exposed for callers outside of the nessie client.